cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/deploy/pipeline/SystemMetadataWriterStage.java

```diff
@@ -45,7 +48,10 @@ public void process(ApplicationWithPrograms input) throws Exception {
     // add system metadata for apps
     ApplicationId appId = input.getApplicationId();
     ApplicationSpecification appSpec = input.getSpecification();
-    SystemMetadataWriter appSystemMetadataWriter = new AppSystemMetadataWriter(metadataStore, appId, appSpec);
+    // only update creation time if this is a new app
+    Map<String, String> properties = metadataStore.getProperties(MetadataScope.SYSTEM, appId);
+    SystemMetadataWriter appSystemMetadataWriter =
+      new AppSystemMetadataWriter(metadataStore, appId, appSpec, !properties.isEmpty());
```

Comment:
this one is for apps right? The check for apps can't be improved unless we make LocalApplicationManager aware of whether its a create or an update. I am not sure if metadata update warrants that change in assumption.